spec/system/admin/payments_stripe_spec.rb

```diff
@@ -20,7 +20,13 @@
 
   before { setup_stripe }
 
-  context "making a new Stripe payment", js: true do
+  after do
+    allow(Spree::Config).to receive(:stripe_connect_enabled).and_return(false)
```

Comment:
I don't think that you need this line. It's only effective before test code, not after. Any `allow` and `expect` mocks should be forgotten after each test case.

Reply:
I noticed that we have the `setup_stripe` helper above. We should have a `teardown_stripe` method to call afterwards with this code.

Reply:
But I'm wondering if we can make this whole `after` callback obsolete by change `setup_stripe` to use `allow` in which case the cleanup is handled automatically. It was introduced in 04ff4fa47cde6799303d007c7aee23c7c1e40bcb but unfortunately the commit message only describes the "what" and not the "why". :disappointed: I think it's worth trying that.

Reply:
No, I meant:

```rb
allow(Stripe).to receive(:api_key).and_return("sk_test_12345")
```

But there may be reason why this wasn't used in the past. Maybe Stripe internals access the values differently. :shrug: :question:

Reply:
> `allow(Stripe).to receive(:api_key).and_return("sk_test_12345"`)

Yeah, I've tried this way as well, but it seemed not to work either... I'm not sure why.

My best take on this is to create a helper method `clear_stripe`, and to call it on each spec in which `stripe_setup` is called. Also found other specs which were leaking the stripe configuration.